.github/workflows/ci.yml

```diff
@@ -0,0 +1,125 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - '[0-9]+.x'
+      - '[0-9]+.[0-9]+.x'
+  pull_request:
+    types: [opened, reopened, synchronize]
+    branches:
+      - master
+      - '[0-9]+.x'
+      - '[0-9]+.[0-9]+.x'
+
+jobs:
+  lint:
+    name: Lint
+    timeout-minutes: 2
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Brightspace/third-party-actions@actions/checkout
+      - uses: Brightspace/third-party-actions@actions/setup-node
+      - uses: actions/cache@v2
```

Comment:
This will need to be added to `Brightspace/third-party-actions` if caching is something we want. @j3parker what do you think?

Reply:
Sounds good

Reply:
What does it cause to be cached?

Reply:
This is configured to cache everything in `node_modules`, which saves roughly 2 minutes from each job in here (2 minutes times 5 jobs = 10 minutes), plus from the release workflow as well.

It will use a hash of `package-lock.json` to bust the cache, which means that we'd need to start using a lock file. But we can automate the lock file updates similar to how we do that in BSI, minimizing its impact on us. Bonus here would be that we'd have a lock file in release branches, which is nice to manage dependencies.

Reply:
More information:
[Caching dependencies to speed up workflows](https://docs.github.com/en/free-pro-team@latest/actions/guides/caching-dependencies-to-speed-up-workflows).